cluster/query_executor.go

```diff
@@ -481,42 +482,29 @@ func (e *QueryExecutor) executeSelectStatement(stmt *influxql.SelectStatement, c
 	}
 
 	// Emit rows to the results channel.
-	var writeN int64
-	var emitted bool
-	for {
-		row := em.Emit()
-		if row == nil {
-			// Check if the query was interrupted while emitting.
-			select {
-			case <-closing:
-				return influxql.ErrQueryInterrupted
-			default:
-			}
-			break
-		}
-
-		result := &influxql.Result{
-			StatementID: statementID,
-			Series:      []*models.Row{row},
-		}
+	var emstats emitterStats
+	emitterCh := make(chan emitterStats, 1)
+	done := make(chan struct{})
+	go e.monitorEmitter(em, stmt, statementID, emitterCh, done, results, closing)
+	defer func() { <-done }()
```

Comment:
There's still more that needs to be added to `QueryManager` to make this work:

1. Allow interrupts to send errors.
2. Attach iterators to `QueryTaskInfo` (these don't get created until after the query is attached though)

I understand that the query manager is supposed to "manage" queries and the executor is supposed to execute them but that's not how it works in the code. The executor controls the query for the entire lifetime of the query. There is not a clear distinction of responsibilities between the executor and manager and I think they should be merged.